File: backend-go/internal/model/state.go
package model

// State represents an issue state (e.g., Backlog, Todo, In Progress, Done).
type State struct {
	BaseModel

	Name        string `gorm:"size:255;not null" json:"name"`
	Color       string `gorm:"size:50;default:#6B7280" json:"color"`
	Group       string `gorm:"size:50;default:backlog" json:"group"`
	Sequence    int    `gorm:"default:1" json:"sequence"`
	IsDefault   bool   `gorm:"default:false" json:"is_default"`
	IsActive    bool   `gorm:"default:true" json:"is_active"`
	ProjectID   uint64 `gorm:"not null;index" json:"project_id"`
	WorkspaceID uint64 `gorm:"not null" json:"workspace_id"`

	// Relationships
	Project            Project           `gorm:"foreignKey:ProjectID" json:"-"`
	Issues             []Issue           `gorm:"foreignKey:StateID" json:"-"`
	SourceTransitions  []StateTransition `gorm:"foreignKey:SourceStateID" json:"-"`
	TargetTransitions  []StateTransition `gorm:"foreignKey:TargetStateID" json:"-"`
}

func (State) TableName() string {
	return "states"
}

// StateTransition represents a workflow transition between states.
// Minimal model for Phase 1 - full workflow comes in Phase 2.
type StateTransition struct {
	BaseModel

	Name          string  `gorm:"size:255;not null" json:"name"`
	Description   *string `gorm:"size:500" json:"description"`
	SourceStateID uint64  `gorm:"not null" json:"source_state_id"`
	TargetStateID uint64  `gorm:"not null" json:"target_state_id"`
	IssueTypeID   *uint64 `json:"issue_type_id"`
	IsAuto        bool    `gorm:"default:false" json:"is_auto"`
	ProjectID     uint64  `gorm:"not null" json:"project_id"`
	WorkspaceID   uint64  `gorm:"not null" json:"workspace_id"`

	SourceState State `gorm:"foreignKey:SourceStateID" json:"-"`
	TargetState State `gorm:"foreignKey:TargetStateID" json:"-"`
}

func (StateTransition) TableName() string {
	return "state_transitions"
}

File: backend/app/models/state.py
from sqlalchemy import String, Integer, ForeignKey, Boolean, BigInteger
from sqlalchemy.orm import Mapped, mapped_column, relationship
from .base import Base, AuditMixin, SoftDeleteMixin

class State(Base, AuditMixin, SoftDeleteMixin):
    __tablename__ = "states"
    
    name: Mapped[str] = mapped_column(String(255), nullable=False)
    color: Mapped[str] = mapped_column(String(50), default="#6B7280")
    group: Mapped[str] = mapped_column(String(50), default="backlog")
    sequence: Mapped[int] = mapped_column(Integer, default=1)
    is_default: Mapped[bool] = mapped_column(Boolean, default=False)
    is_active: Mapped[bool] = mapped_column(Boolean, default=True)
    
    project_id: Mapped[int] = mapped_column(BigInteger, ForeignKey("projects.id"), nullable=False)
    workspace_id: Mapped[int] = mapped_column(BigInteger, ForeignKey("workspaces.id"), nullable=False)
    
    project: Mapped["Project"] = relationship(back_populates="states")
    workspace: Mapped["Workspace"] = relationship()
    issues: Mapped[list["Issue"]] = relationship(back_populates="state")
    
    # 工作流转换关系
    source_transitions: Mapped[list["StateTransition"]] = relationship(
        back_populates="source_state",
        foreign_keys="StateTransition.source_state_id"
    )
    target_transitions: Mapped[list["StateTransition"]] = relationship(
        back_populates="target_state",
        foreign_keys="StateTransition.target_state_id"
    )